tests/deps.edn

```diff
@@ -0,0 +1,78 @@
+; Copyright 2022 Cognitect, Inc.
```

Comment:
So a big part of this change is to create a tests project that contains all the tests; previously there was some very dodgy dependencies that were almost circular. Unfortunately, this also meant moving all the test namespaces into the new project as well.